storage/options.go

```diff
@@ -206,6 +210,16 @@ func (o *options) Validate() error {
 		}
 	}
 
+	// validate that persist manager is present, if not return
+	// error if error occurred during default creation otherwise
+	// it was set to nil by a caller
+	if o.persistManager == nil {
+		if err := o.persistManagerErr; err != nil {
```

Comment:
How about just not setting persistManager by default, and always requiring users to override it. Would be able to avoid this ugliness

Reply:
Sounds good, will do.